packages/react-components/src/components/MessageThread.tsx

```diff
@@ -572,6 +574,20 @@ export type MessageThreadProps = {
    * Optional strings to override in component
    */
   strings?: Partial<MessageThreadStrings>;
+
+  /* @conditional-compile-remove(file-sharing) */
+  /**
+   * @beta
+   * Optional function called when someone clicks on the file download icon.
+   */
+  fileDownloadHandler?: FileDownloadHandler;
+  /* @conditional-compile-remove(file-sharing) */
+  /**
+   * @beta
+   * Property name that contains information about file downloads in `message.metadata` object.
+   * @defaultValue fileSharingMetadata
+   */
+  fileDownloadMetadataKey?: string;
```

Comment:
semantic nit: the file isn't a fileDownload currently, it's a fileAttachment (but would rather keep fileDownload if it drives consistency)

Reply:
Let's see what others have to say about this API and we can revisit the name basis on what path we take :)